lib/web/servers.go

```diff
@@ -127,3 +128,53 @@ func (h *Handler) getDesktopHandle(w http.ResponseWriter, r *http.Request, p htt
 	// to see the desktop once in the UI, so just take the first one.
 	return ui.MakeDesktop(windowsDesktops[0]), nil
 }
+
+// desktopIsActive checks if a desktop has an active session and if so, returns a one time token that is needed
+// to start a session to said desktop.
+//
+// GET /v1/webapi/sites/:site/desktop_is_active/:desktop
+//
+// Response body:
+//
+// {"active": bool}
+func (h *Handler) desktopIsActive(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	desktopName := p.ByName("desktop")
+	trackers, err := h.auth.proxyClient.GetActiveSessionTrackers(r.Context())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	checker, err := ctx.GetUserAccessChecker()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	for _, tracker := range trackers {
+		if tracker.GetSessionKind() == types.WindowsDesktopSessionKind && tracker.GetState() == types.SessionState_SessionStateRunning && tracker.GetDesktopName() == desktopName {
+			desktops, err := h.auth.accessPoint.GetWindowsDesktops(r.Context(), types.WindowsDesktopFilter{Name: tracker.GetDesktopName()})
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			if len(desktops) == 0 {
+				break
+			}
+
+			err = checker.CheckAccess(desktops[0], services.AccessMFAParams{})
+			switch {
+			case err == nil:
+				return desktopIsActive{true}, nil
+			case trace.IsAccessDenied(err):
+				break
+			default:
+				return nil, trace.Wrap(err)
+			}
+		}
+	}
+
+	return desktopIsActive{false}, nil
+}
+
+type desktopIsActive struct {
```

Comment:
nitpick: just to be consistent with other request/response structs
```suggestion
type desktopIsActiveResponse struct {
```